controllers/ssp_controller.go

```diff
@@ -186,7 +206,71 @@ func cleanup(request *common.Request) error {
 	return err
 }
 
+func pauseCRs(sspRequest *common.Request, kinds []string) error {
+	patch := []byte(`{
+  "metadata":{
+    "annotations":{"kubevirt.io/operator.paused": "true"}
+  }
+}`)
+	for _, kind := range kinds {
+		crs := &unstructured.UnstructuredList{}
+		crs.SetKind(kind)
+		crs.SetAPIVersion("ssp.kubevirt.io/v1")
+		err := sspRequest.Client.List(sspRequest.Context, crs)
+		if err != nil {
+			sspRequest.Logger.Error(err, fmt.Sprintf("Error listing %s CRs: %s", kind, err))
+			return err
+		}
+		for _, item := range crs.Items {
+			err = sspRequest.Client.Patch(sspRequest.Context, &item, client.RawPatch(types.MergePatchType, patch))
+			if err != nil {
+				// Patching failed, maybe the CR just got removed? Log an error but keep going.
+				sspRequest.Logger.Error(err, fmt.Sprintf("Error pausing %s from namespace %s: %s",
+					item.GetName(), item.GetNamespace(), err))
+			}
+		}
+	}
+
+	return nil
+}
+
+func listExistingCRDKinds(sspRequest *common.Request) []string {
+	// Get the list of all CRDs and build a list of the SSP ones
+	crds := &unstructured.UnstructuredList{}
+	crds.SetKind("CustomResourceDefinition")
+	crds.SetAPIVersion("apiextensions.k8s.io/v1")
+	err := sspRequest.Client.List(sspRequest.Context, crds)
+	foundKinds := make([]string, 0, len(kvsspCRDs))
+	if err == nil {
+		for _, item := range crds.Items {
+			name := item.GetName()
+			for crd, kind := range kvsspCRDs {
+				if crd == name {
+					foundKinds = append(foundKinds, kind)
+					break
+				}
+			}
+		}
+	}
+
+	return foundKinds
+}
+
 func reconcileOperands(sspRequest *common.Request) ([]common.ResourceStatus, error) {
+	opVer := os.Getenv("OPERATOR_VERSION")
+	if opVer != "" {
+		operatorVersion = opVer
+	}
```

Comment:
The variable `operatorVersion` is accessed before this block is executed, in function `preUpdateStatus`.
You could extract this block into a separate function and use it in all needed places.
```
func getOperatorVersion() string {
	opVer := os.Getenv("OPERATOR_VERSION")
	if opVer != "" {
		return "devel"
	}
	return opVer
}
```

The function `os.Getenv()` caches env variables in a map, so it is not expensive to call it on each reconcile iteration.